src/hardware/ne2000.cpp

```diff
@@ -586,20 +586,21 @@ bx_ne2k_c::page0_read(io_port_t offset, io_width_t io_len)
 void
 bx_ne2k_c::page0_write(io_port_t offset, io_val_t data, io_width_t io_len)
 {
-  auto value = check_cast<uint8_t>(data);
-  unsigned int new_value = 0;
-
   BX_DEBUG("page 0 write to port %04x, len=%u", (unsigned)offset, (unsigned)io_len);
 
   // It appears to be a common practice to use outw on page0 regs...
 
   // break up outw into two outb's
   if (io_len == io_width_t::word) {
+    auto value = check_cast<uint16_t>(data);
     page0_write(offset, (value & 0xff), io_width_t::byte);
     page0_write(offset + 1, ((value >> 8) & 0xff), io_width_t::byte);
     return;
   }
 
+  auto value = check_cast<uint8_t>(data);
```

Comment:
`const auto`

Reply:
I don't think I can safely make the lower one `const`. Later in the function:
````cpp
case 0x7:  // ISR
    value &= 0x7f;  // clear RST bit - status-only bit
```

Reply:
First `value` made `const auto`.

Reply:
> I don't think I can safely make the lower one const. Later in the function

Roger that (and sorry for the false alarm!)